packages/@aws-cdk/aws-batch/lib/ecs-container-definition.ts

```diff
@@ -0,0 +1,939 @@
+  }
+
+  /**
+   * Creates a Host volume. This volume will persist on the host at the specified `hostPath`.
+   * If the `hostPath` is not specified, Docker will choose the host path. In this case,
+   * the data may not persist after the containers that use it stop running.
+   */
+  static host(options: HostVolumeOptions) {
+    return new HostVolume(options);
+  }
+
+  /**
+   * The name of this volume
+   */
+  public readonly name: string;
+
+  /**
+   * The path on the container that this volume will be mounted to
+   */
+  public readonly containerPath: string;
+
+  /**
+   * Whether or not the container has readonly access to this volume
+   *
+   * @default false
+   */
+  public readonly readonly?: boolean;
+
+  constructor(options: EcsVolumeOptions) {
+    this.name = options.name;
+    this.containerPath = options.containerPath;
+    this.readonly = options.readonly;
+  }
+}
+
+/**
+ * Options for configuring an EfsVolume
+ */
+export interface EfsVolumeOptions extends EcsVolumeOptions {
+  /**
+   * The EFS File System that supports this volume
+   */
+  readonly fileSystem: IFileSystem;
+
+  /**
+   * The directory within the Amazon EFS file system to mount as the root directory inside the host.
+   * If this parameter is omitted, the root of the Amazon EFS volume is used instead.
+   * Specifying `/` has the same effect as omitting this parameter.
+   * The maximum length is 4,096 characters.
+   *
+   * @default - root of the EFS File System
+   */
+  readonly rootDirectory?: string;
+
+  /**
+   * Enables encryption for Amazon EFS data in transit between the Amazon ECS host and the Amazon EFS server
+   *
+   * @see https://docs.aws.amazon.com/efs/latest/ug/encryption-in-transit.html
+   *
+   * @default false
+   */
+  readonly enableTransitEncryption?: boolean;
+
+  /**
+   * The port to use when sending encrypted data between the Amazon ECS host and the Amazon EFS server.
+   * The value must be between 0 and 65,535.
+   *
+   * @see https://docs.aws.amazon.com/efs/latest/ug/efs-mount-helper.html
+   *
+   * @default - chosen by the EFS Mount Helper
+   */
+  readonly transitEncryptionPort?: number;
+
+  /**
+   * The Amazon EFS access point ID to use.
+   * If an access point is specified, `rootDirectory` must either be omitted or set to `/`
+   * which enforces the path set on the EFS access point.
+   * If an access point is used, `enableTransitEncryption` must be `true`.
+   *
+   * @see https://docs.aws.amazon.com/efs/latest/ug/efs-access-points.html
+   *
+   * @default - no accessPointId
+   */
+  readonly accessPointId?: string;
+
+  /**
+   * Whether or not to use the AWS Batch job IAM role defined in a job definition when mounting the Amazon EFS file system.
+   * If specified, `enableTransitEncryption` must be `true`.
+   *
+   * @see https://docs.aws.amazon.com/batch/latest/userguide/efs-volumes.html#efs-volume-accesspoints
+   *
+   * @default false
+   */
+  readonly useJobDefinitionRole?: boolean;
```

Comment:
Batch calls this role the `JobRole[Arn]` , Batch doesn't really use the term JobDefinitionRole in its docs.